docs/tutorials/tpu-cluster-setup.md

```diff
@@ -0,0 +1,135 @@
+# Setting up a TPU Cluster
+
+This guide will walk you through the steps to set up and manage a TPU cluster using Google Cloud Platform (GCP) and Ray.
+
+## Prerequisites
+
+Before you begin, ensure you have:
+- Completed the [basic installation](installation.md)
+- Access to a Google Cloud project with TPU quota
+- [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) installed
+- [Ray](https://docs.ray.io/en/latest/installation.html) installed
+
+## TPU Cluster Architecture
+
+A typical TPU cluster consists of:
+- **Head Node**: A persistent GCP VM that coordinates the cluster
+- **Worker Nodes**: Autoscaling TPU VMs that execute the actual computation
+
+## Cluster Configuration
+
+The cluster configuration is defined in a [YAML template file](../../infra/marin-cluster-template.yaml). This template allows you to customize various aspects of your cluster:
+
+- Machine types for head and worker nodes
+- Number of CPUs and amount of RAM
+- Disk size and type
+- TPU types and versions
+- Minimum and maximum worker counts
+- Docker images and initialization commands
+- Network configuration
+- Environment variables and secrets
+
+For our clusters, each config is in a separate file in the `infra` directory. These files are automatically generated by the
+`infra/update-cluster-configs.py` script, which reads the `infra/marin-cluster-template.yaml` file.
+More information can be found in the [infra/README.md](../../infra/README.md).
```

Comment:
Can we move this readme into docs?

Reply:
Hmmm That Readme is pretty long and has internal details like our TPU types and stuff. Maybe I can make another TPU-cluster-setup-internal.md and put that ? @dlwh What do you think ?

Reply:
sure

Reply:
Will address this in another PR.